c10/core/TensorImpl.cpp

```diff
@@ -113,6 +113,47 @@ bool TensorImpl::has_storage() const {
   return storage_;
 }
 
+void TensorImpl::set_memory_format_tag(MemoryFormat memory_format) {
+  AT_ASSERT(
+      !is_variable()); // TODO: remove this when Variable and Tensor are merged
+  memory_format_tag_ = memory_format;
+}
+
+bool TensorImpl::maybe_as_channels_last() {
```

Comment:
In the design proposal, it seemed like a lot of people were confused about exactly how the permutation here operates. I think a Note, capturing some of the discussion from the design proposal, would help subsequent people trying to understand this code a lot.